numpy/lib/function_base.py

```diff
@@ -2950,15 +2950,8 @@ def i0(x):
     array([ 1.00000000+0.j        ,  0.18785373+0.64616944j])
 
     """
-    x = atleast_1d(x).copy()
-    y = empty_like(x)
-    ind = (x < 0)
-    x[ind] = -x[ind]
-    ind = (x <= 8.0)
-    y[ind] = _i0_1(x[ind])
-    ind2 = ~ind
-    y[ind2] = _i0_2(x[ind2])
-    return y.squeeze()
+    x = np.abs(x)
+    return piecewise(x, [x <= 8.0], [_i0_1, _i0_2])
```

Comment:
Use `np.where`

Reply:
Nevermind, what you have is better.

Reply:
Yes, otherwise both functions get called for all `x`.